components/suggestion/suggestion_date.jsx

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import $ from 'jquery';
+import PropTypes from 'prop-types';
+import React from 'react';
+import {Popover} from 'react-bootstrap';
+import ReactDOM from 'react-dom';
+
+import SuggestionStore from 'stores/suggestion_store.jsx';
+
+export default class SuggestionDate extends React.Component {
+    static propTypes = {
+        suggestionId: PropTypes.string.isRequired,
+        location: PropTypes.string,
+        renderDividers: PropTypes.bool,
+        onCompleteWord: PropTypes.func.isRequired,
+    };
+
+    static defaultProps = {
+        renderDividers: false,
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.getStateFromStores = this.getStateFromStores.bind(this);
+
+        this.getContent = this.getContent.bind(this);
+
+        this.handleSuggestionsChanged = this.handleSuggestionsChanged.bind(this);
+
+        this.state = this.getStateFromStores(props.suggestionId);
+    }
+
+    getStateFromStores(suggestionId) {
+        const suggestions = SuggestionStore.getSuggestions(suggestionId || this.props.suggestionId);
+
+        return {
+            matchedPretext: suggestions.matchedPretext,
+            items: suggestions.items,
+            terms: suggestions.terms,
+            components: suggestions.components,
+            selection: suggestions.selection,
+        };
+    }
+
+    componentDidMount() {
+        SuggestionStore.addSuggestionsChangedListener(this.props.suggestionId, this.handleSuggestionsChanged);
+    }
+
+    componentWillUnmount() {
+        SuggestionStore.removeSuggestionsChangedListener(this.props.suggestionId, this.handleSuggestionsChanged);
+    }
+
+    getContent() {
```

Comment:
We can remove that. It's copied from the SuggestionList component, but that only needs it because of some code that it shares with the SearchSuggestionList, but the SuggestionDate component is completely independent of those two